src/cdk/overlay/position/flexible-connected-position-strategy.ts

```diff
@@ -528,14 +540,26 @@ export class FlexibleConnectedPositionStrategy implements PositionStrategy {
     return this;
   }
 
+  /**
+   * Sets a custom element to use as the host for the popover.
+   * The popover will be inserted after this element in the DOM.
+   * If null, the overlay will be inserted after the origin.
+   * @param element The element to use as the host for the popover.
+   */
+  withCustomPopoverHostElement(element: FlexibleConnectedPositionStrategyOrigin | null): this {
```

Comment:
Rather than introducing a new API, couldn't we control this through the `getPopoverInsertionPoint`? E.g. we can have it return something like `Element | null | {parent: Element, insertAs: 'child'}`